components/cli/src/main/java/datadog/cli/CLIHelper.java

```diff
@@ -9,18 +9,102 @@
 import java.nio.file.Paths;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 
+/** Helper class for retrieving and parsing JVM arguments. */
 public final class CLIHelper {
   private static final List<String> VM_ARGS = findVmArgs();
+  private static Map<String, String> VM_ARGS_MAP = Collections.emptyMap();
+  private static int currentIndex = 0; // Tracks the last processed index in VM_ARGS
 
   public static List<String> getVmArgs() {
     return VM_ARGS;
   }
 
+  public static String getArgValue(String key) {
+    int numArgs = VM_ARGS.size();
+    // Lazy population of cache
+    synchronized (VM_ARGS_MAP) {
+      // Double check after acquiring lock
+      if (VM_ARGS_MAP.containsKey(key)) {
+        return VM_ARGS_MAP.get(key);
+      }
+      if (currentIndex >= numArgs) {
+        return null;
+      }
+
+      // Initialize cache if empty
+      if (VM_ARGS_MAP.isEmpty()) {
+        VM_ARGS_MAP = new HashMap<>();
+      }
+
+      // Process remaining args
+      while (currentIndex < numArgs) {
+        populateCache(VM_ARGS.get(currentIndex));
+        currentIndex++;
+
+        // Check if we found our key
+        if (VM_ARGS_MAP.containsKey(key)) {
+          return VM_ARGS_MAP.get(key);
```

Comment:
As above, since we know we're not caching `null` values we can avoid the extra lookup:
```suggestion
        value = VM_ARGS_MAP.get(key);
        if (value != null) {
          return value;
```